### PR TITLE
deploy contracts as plain code

### PR DESCRIPTION
**Pull request by kroggen**

### CODE REVIEW

**Comment by kroggen** [comment]
@hayarobi This is one of the most important PRs

**Comment by hayarobi** [comment]
Does this path prevent deploying compiled code?Is it breaking backward compatibility?

File: contract/vm.go [hunk 1]
```diff
@@ -891,17 +891,32 @@ func setRandomSeed(ctx *vmContext) {
...
+func setContract(contractState *statedb.ContractState, contractAddress, payload []byte, ctx *vmContext) ([]byte, []byte, error) {
 	codePayload := luacUtil.LuaCodePayload(payload)
 	if _, err := codePayload.IsValidFormat(); err != nil {
 		ctrLgr.Warn().Err(err).Str("contract", types.EncodeAddress(contractAddress)).Msg("deploy")
 		return nil, nil, err
 	}
 	code := codePayload.Code()
+
+	// if hardfork version 4
+	if ctx.blockInfo.ForkVersion >= 4 {
```
Comment (shepelt) [inline contract/vm.go:903]: I think this part would insure that code running in blocks with forkversion < 4 to run as previously @hayarobi

Comment (kroggen) [reply contract/vm.go:903]: The contract execution code was not modified. It remains exactly the same

**Comment by kroggen** [comment]
@hayarobi please finish review

**Review by hayarobi — APPROVED** [review]
lgtm

**Pull request merged**